src/main/java/analyzer/solutions/OptimalFormat.java

```diff
@@ -0,0 +1,62 @@
+package analyzer.solutions;
+
+import com.github.javaparser.ast.expr.*;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.Node;
+import static com.github.javaparser.JavaParser.*;
+import java.io.File;
+import com.github.javaparser.JavaParser;
+import java.util.List;
+import com.github.javaparser.ast.visitor.VoidVisitorAdapter;
+import com.github.javaparser.ast.stmt.IfStmt;
+import com.github.javaparser.ast.stmt.ReturnStmt;
+import com.github.javaparser.ast.visitor.TreeVisitor;
+import java.util.ArrayList;
+
+
+public class OptimalFormat {
+
+    public static CompilationUnit getContent(String flname) throws Exception{
+        return JavaParser.parse(new File(flname));
+    }
+
+
+    // visits the method call expressions and adds them to a List
+    private static class OptionalMethodVisitor extends VoidVisitorAdapter<List<MethodCallExpr>> {
+        @Override
+        public void visit(MethodCallExpr n, List<MethodCallExpr> arg) {
+            arg.add(n);
+            super.visit(n, arg);
+        }
+    }
+
+    // Checks the return for the correct statement (optimal solution 1)
+    public static boolean checkReturn(List<ReturnStmt> returns){
+        if(returns.size() > 1){
+            System.out.println("Can be refactored to use less return statements");
+            return false;
+        }
+        else{
+            ReturnStmt n = returns.get(0);
+            List<MethodCallExpr> returnMethods = new ArrayList<MethodCallExpr>();
+            n.accept(new OptionalMethodVisitor(), returnMethods);
+            if(returnMethods.size() == 1){
+                if(returnMethods.get(0).getScope().get().toString().equals("String") &&
+                        returnMethods.get(0).getNameAsString().equals("format")){
+                    if(!returnMethods.get(0).getArgument(0).toString().equals("\"One for %s, one for me.\"")){
+                        return false;
+                    };
+                    if(!returnMethods.get(0).getArgument(1).toString().equals("name == null ? \"you\" : name") &&
+                        !returnMethods.get(0).getArgument(1).toString().equals("name != null ? name : \"you\"")) {
+                        return false;
+                    }
+                }
+            }
+            else {
+                return false;
+            }
+            return true;
```

Comment:
Unclear what this last return true is supposed to link with. The if-else chain looks odd. Consider refactoring to a guard clause where we first check if the return methods size is NOT equal to 1 and return false immediately. Then we can put everything else at the top level.

```java
// Guard clause
if (returnMethods.size() != 1) return false;
MethodCallExpr returnMethod = returnMethods.get(0);

// Another guard clause
if (!returnMethod.getScope().get().toString().equals("String")
  || !returnMethod.getNameAsString().equals("format")) return false;

// And another
if (!returnMethod.getArgument(0).toString().equals("...")) return false;
// NOTE: I don't actually know the return type from getArgument, just assuming it is "Argument"
Argument secondArgument = returnMethod.getArgument(1);

// Another
if (!secondArgument.toString().equals(...)
  && !secondArgument.toString().equals(...)) return false;

// Otherwise
return true;
```